src/sqlancer/cnosdb/ast/CnosDBAggregate.java

```diff
@@ -0,0 +1,112 @@
+package sqlancer.cnosdb.ast;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import sqlancer.Randomly;
+import sqlancer.cnosdb.CnosDBSchema.CnosDBDataType;
+import sqlancer.cnosdb.ast.CnosDBAggregate.CnosDBAggregateFunction;
+import sqlancer.common.ast.FunctionNode;
+
+public class CnosDBAggregate extends FunctionNode<CnosDBAggregateFunction, CnosDBExpression>
+        implements CnosDBExpression {
+
+    public CnosDBAggregate(List<CnosDBExpression> args, CnosDBAggregateFunction func) {
+        super(func, args);
+    }
+
+    public enum CnosDBAggregateFunction {
+        AVG(CnosDBDataType.DOUBLE),
+        MAX(CnosDBDataType.DOUBLE, CnosDBDataType.INT, CnosDBDataType.STRING, CnosDBDataType.TIMESTAMP,
+                CnosDBDataType.UINT),
+        MIN(CnosDBDataType.DOUBLE, CnosDBDataType.INT, CnosDBDataType.STRING, CnosDBDataType.TIMESTAMP,
+                CnosDBDataType.UINT),
+        COUNT(CnosDBDataType.INT) {
+            @Override
+            public CnosDBDataType[] getInputTypes(CnosDBDataType returnType) {
+                return new CnosDBDataType[] { CnosDBDataType.getRandomType() };
+            }
+        },
+        SUM(CnosDBDataType.INT, CnosDBDataType.DOUBLE, CnosDBDataType.UINT), APPROX_MEDIAN(CnosDBDataType.DOUBLE);
+
+        // Currently these aggregate functions have bugs https://github.com/cnosdb/cnosdb/issues/786
```

Comment:
As an optional suggestion, I think it would be more elegant to include the currently commented-out code. If, for example, you require a random aggregate function, you could still remove the buggy aggregate functions from the selection. The general pattern is described at https://github.com/sqlancer/sqlancer/blob/67751dc3d83b880cd8c7ed53416b8b157a2ba2ea/CONTRIBUTING.md#implementing-support-for-a-new-dbms, and I see you added a `CnosDBBugs` class already.

Reply:
Thanks. I removed the comment and applied that pattern.